src/main/java/com/google/sps/data/JobJSON.java

```diff
@@ -79,4 +81,128 @@ public JobJSON(String projectId, String name, String id, String type, String sdk
     this.price = price;
     this.sdkName = sdkName;
   }
+  
+  // Helper function for the equals method
+  // Returns true if the 2 fields are different
+  private boolean notEqualsField(Double f1, Double f2) {
+    if ((f1 == null && f2 != null) ||
+        (f1 != null && f2 == null) ||
+        (f1 != null && f2 != null && f1.compareTo(f2) != 0)) {
+      return true;
+    }
+
+    return false;
+  }
+  
+  // Helper function for the equals method
+  // Returns true if the 2 fields are different
+  private boolean notEqualsField(Integer f1, Integer f2) {
+    if ((f1 == null && f2 != null) ||
+        (f1 != null && f2 == null) ||
+        (f1 != null && f2 != null && f1.compareTo(f2) != 0)) {
+      return true;
+    }
+
+    return false;
+  }
+  
+  // Helper function for the equals method
+  // Returns true if the 2 fields are different
+  private boolean notEqualsField(String f1, String f2) {
+    if ((f1 == null && f2 != null) ||
+        (f1 != null && f2 == null) ||
+        (f1 != null && f2 != null && f1.compareTo(f2) != 0)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof JobJSON)) {
+      return false;
+    }
+    JobJSON job2 = (JobJSON) obj;
+
+    if (notEqualsField(this.id, job2.id)) {
+      return false;
+    }
+    if (notEqualsField(this.projectId, job2.projectId)) {
+      return false;
+    }
+    if (notEqualsField(this.name, job2.name )) {
+      return false;
+    }
+    if (notEqualsField(this.type , job2.type)) {
+      return false;
+    }
+    if (notEqualsField(this.sdk, job2.sdk)) {
+      return false;
+    }
+    if (notEqualsField(this.sdkSupportStatus, job2.sdkSupportStatus)) {
+      return false;
+    }
+    if (notEqualsField(this.region, job2.region)) {
+      return false;
+    }
+    if (this.currentWorkers != job2.currentWorkers) {
```

Comment:
Was there a particular reason for not using the notEqualsField function here?

Reply:
currentWorkers is an int, not an Integer, so it can't have assigned a null value. Basically, the function is designed to check whether the fields are equal or if they are both null (in which case they are still equal, but can't really find this out by calling compareTo, as it raises NullPointException). For an int, it is sufficient to just check if they are equal or not.

Reply:
Ah I understand now! Thank you :D